apiserver/shared.go

```diff
@@ -9,13 +9,22 @@ import (
 	"github.com/juju/collections/set"
 	"github.com/juju/errors"
 	"github.com/juju/loggo"
-	"github.com/juju/pubsub"
 
 	"github.com/juju/juju/core/presence"
+	"github.com/juju/juju/feature"
+	"github.com/juju/juju/pubsub/apiserver"
 	"github.com/juju/juju/pubsub/controller"
 	"github.com/juju/juju/state"
 )
 
+// SharedHub represents the methods of the pubsub.StructuredHub
+// that are used. The context uses an interface to allow mocking
+// of the hub.
+type SharedHub interface {
```

Comment:
This is fun - now any facade can bounce the API server! (Not a serious complaint but, you know...)

Reply:
We can probably remove this when whe presence code is gone :)